firedrake/mesh.py

```diff
@@ -1787,50 +1793,21 @@ def input_ordering_sf(self):
         """
         if not isinstance(self.topology, VertexOnlyMeshTopology):
             raise AttributeError("Input ordering is only defined for vertex-only meshes.")
-        sf = PETSc.SF().create(comm=self.comm)
         nroots = self.input_ordering.num_cells()
-        input_ranks = self.topology_dm.getField("inputrank")
-        self.topology_dm.restoreField("inputrank")
-        input_indices = self.topology_dm.getField("inputindex")
-        self.topology_dm.restoreField("inputindex")
-        nleaves = len(input_ranks)
-        input_ranks_and_idxs = np.empty(2 * nleaves, dtype=IntType)
-        input_ranks_and_idxs[0::2] = input_ranks
-        input_ranks_and_idxs[1::2] = input_indices
-        # local looks like the below, which means we can just pass in None
-        # local = numpy.arange(nleaves, dtype=IntType)
-        sf.setGraph(nroots, None, input_ranks_and_idxs)
-        return sf
+        e_p_map = self.cell_closure[:, -1]
+        ilocal = np.empty_like(e_p_map)
+        if len(e_p_map) > 0:
+            cStart = e_p_map.min()
+            ilocal[e_p_map - cStart] = np.arange(len(e_p_map))
+        return VertexOnlyMeshTopology._make_input_ordering_sf(self.topology_dm, nroots, ilocal)
 
     @utils.cached_property  # TODO: Recalculate if mesh moves
     def input_ordering_without_halos_sf(self):
         """
         Return a PETSc SF which has :func:`~.VertexOnlyMesh` input ordering
         vertices as roots and this mesh's non-halo vertices as leaves.
         """
-        if not isinstance(self.topology, VertexOnlyMeshTopology):
-            raise AttributeError("Input ordering is only defined for vertex-only meshes.")
-        sf = PETSc.SF().create(comm=self.comm)
-        nroots = self.input_ordering.num_cells()
-        ranks = self.topology_dm.getField("DMSwarm_rank")
-        self.topology_dm.restoreField("DMSwarm_rank")
-        input_ranks = self.topology_dm.getField("inputrank")
-        self.topology_dm.restoreField("inputrank")
-        input_index = self.topology_dm.getField("inputindex")
-        self.topology_dm.restoreField("inputindex")
-        # only include leaves where points are on this rank. This will exclude
-        # any points where the point was not found on the mesh.
-        idxs_to_include = ranks == self.comm.rank
-        input_ranks = input_ranks[idxs_to_include]
-        input_indices = input_index[idxs_to_include]
-        nleaves = len(input_ranks)
-        input_ranks_and_idxs = np.empty(2 * nleaves, dtype=IntType)
-        input_ranks_and_idxs[0::2] = input_ranks
-        input_ranks_and_idxs[1::2] = input_indices
-        # local looks like the below, which means we can just pass in None
-        # local = numpy.arange(nleaves, dtype=IntType)
-        sf.setGraph(nroots, None, input_ranks_and_idxs)
-        return sf
+        return self.input_ordering_sf.createEmbeddedLeafSF(np.arange(self.cell_set.size, dtype=IntType))
```

Comment:
Nice! I hadn't realised you could do this.

Reply:
I think this relies on the assumption that halos are at the end of the leaves, am I correct? If so you should probably comment that here.

Reply:
Yes. Done!